### PR TITLE
fix postgresql service account tpl and config

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

fix postgresql service account tpl and config

## Related Issues

- https://github.com/astronomer/issues/issues/6747
- https://github.com/astronomer/issues/issues/6805

## Testing

QA should able to customize posgresql service account 

## Merging

cherry-pick to release-0.36

### CODE REVIEW

**Review by rishkarajgi — CHANGES_REQUESTED** [review]
Please create a ticket with relevant details and update the PR description

**Review by rujhan-arora-astronomer — APPROVED** [review]
Looks fine other than test. Please complete it before merging

File: charts/postgresql/templates/_helpers.tpl [hunk 1]
```diff
@@ -303,3 +303,12 @@ Return  the proper Storage Class
 {{- define "postgresql.storageClass" -}}
 storageClassName: {{ or .Values.persistence.storageClass .Values.global.storageClass | default "" }}
 {{- end -}}
+
```
Comment (danielhoherd) [inline charts/postgresql/templates/_helpers.tpl:306]:
```suggestion
```

File: charts/postgresql/templates/_helpers.tpl [hunk 2]
```diff
@@ -303,3 +303,12 @@ Return  the proper Storage Class
...
+
+{{ define "postgresql.serviceAccountName" -}}
```
[the remark this answers is not in the document]
Comment (danielhoherd) [reply charts/postgresql/templates/_helpers.tpl:308]: They are covered by this test that dynamically finds all pod manager templates and ensures that serviceAccountName is either `default` or `release-name-*` https://github.com/astronomer/astronomer/blob/master/tests/chart_tests/test_byo_sa.py#L137-L161

**Review by danielhoherd — APPROVED** [review]
LGTM.

**Review by rishkarajgi — APPROVED** [review]

**Pull request merged**